src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,69 @@
-export const Pagination = () => {};
+import React from 'react';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}: Props) => {
+  const numOfPages = Math.ceil(total / perPage);
+  const pages = Array.from({ length: numOfPages }, (_, i) => i + 1);
+  const prevHandler = () => currentPage !== 0 && onPageChange(currentPage - 1);
```

Comment:
This is a bug. Your `currentPage` is 1-indexed, but here you are treating it as 0-indexed. You should check if `currentPage` is not equal to 1 instead of 0.